spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -1272,6 +1291,14 @@ private void logConsumerException(Throwable t) {
 				logger.warn("Consumer raised exception, processing can restart if the connection factory supports it. "
 						+ "Exception summary: " + t);
 			}
+			publishEvent("Consumer raised exception, attempting restart", false, t);
+		}
+
+		private void publishEvent(String reason, boolean fatal, Throwable t) {
```

Comment:
I don't have the gold ground answer, but looks like this method 3is enough specific just to those exceptions above.
So, I've changed my mind and merge it as is.
If we find other places to emit events, we always will be able to introduce other top-level methods or just use `applicationEventPublisher` directly.